go/teams/handler.go

```diff
@@ -70,3 +70,79 @@ func HandleChangeNotification(ctx context.Context, g *libkb.GlobalContext, rows
 	}
 	return nil
 }
+
+func HandleSBSRequest(ctx context.Context, g *libkb.GlobalContext, msg keybase1.TeamSBSMsg) error {
+	for _, invitee := range msg.Invitees {
+		if err := handleSBSSingle(ctx, g, msg.TeamID, invitee); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func handleSBSSingle(ctx context.Context, g *libkb.GlobalContext, teamID keybase1.TeamID, invitee keybase1.TeamInvitee) error {
+	uv := NewUserVersion(invitee.Uid, invitee.EldestSeqno)
+	req, err := reqFromRole(uv, invitee.Role)
+	if err != nil {
+		return err
+	}
+	req.CompletedInvites = make(map[keybase1.TeamInviteID]keybase1.UID)
+	req.CompletedInvites[invitee.InviteID] = invitee.Uid
+
+	team, err := Load(ctx, g, keybase1.LoadTeamArg{
+		ID:          teamID,
+		ForceRepoll: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// verify the invite info:
+
+	// find the invite in the team chain
+	invite, found := team.chain().FindActiveInviteByID(invitee.InviteID)
+	if !found {
+		return libkb.NotFoundError{}
+	}
+	category, err := invite.Type.C()
+	if err != nil {
+		return err
+	}
+	switch category {
+	case keybase1.TeamInviteCategory_SBS:
+		//  resolve assertion in link
+		ityp, err := invite.Type.String()
+		if err != nil {
+			return err
+		}
+		assertion := string(invite.Name) + "@" + ityp
+
+		res := g.Resolver.ResolveWithBody(assertion)
+		if res.GetError() != nil {
+			return res.GetError()
+		}
+		uid, err := g.GetUPAKLoader().LookupUID(ctx, res.GetNormalizedUsername())
+		if err != nil {
+			return err
+		}
+
+		// check resolved assertion uid with invitee.Uid
+		if uid != invitee.Uid {
+			return fmt.Errorf("chain sbs assertion %s resolved to uid %s, which doesn't match uid %s in team.sbs message", assertion, uid, invitee.Uid)
+		}
+	case keybase1.TeamInviteCategory_EMAIL:
+		// nothing to verify, need to trust the server
```

Comment:
Can you still verify: (1) that there's an existing, active by-email invitation; and (2) that the invitation ID matches?  I think this is important so that way if you've never invited someone by email, your client will never allow this feature.